pkg/etcdmgr/bootstrap/discovery.go

```diff
@@ -78,10 +84,9 @@ func queryDiscoveryService(token string) (*store.Event, error) {
 			if resp.StatusCode == http.StatusOK {
 				break
 			}
-			resp.Body.Close()
```

Comment:
you are right, good catch.  i think the anonymous function defined for the defer statement could be refactored out to a useful standalone function of SafeHttpResponseClose() or something like that.  Then it could be called inline in the for loop and as a defer statement catchall for the entire function too.